cinema/views.py

```diff
@@ -56,3 +88,50 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action == "list":
+            queryset = (
+                queryset
+                .select_related("movie", "cinema_hall")
+                .annotate(
+                    tickets_available=(
+                        F("cinema_hall__rows") * F("cinema_hall__seats_in_row")
+                    ) - Count("tickets")
+                )
+            )
+            movie = self.request.query_params.get("movie")
+            date = self.request.query_params.get("date")
+            if movie:
+                queryset = queryset.filter(movie__id=movie)
+
+            if date:
+                queryset = queryset.filter(show_time__date=date)
+
+        if self.action in "retrieve":
```

Comment:
should be `==` Right now it will try to match self.action vs every character in a string